src/nested_pandas/nestedframe/core.py

```diff
@@ -916,6 +916,127 @@ def dropna(
+            Always False when applied to nested layers.
+        key : callable, optional
+            Apply the key function to the values before sorting.
+
+        Returns:
+        --------
+        DataFrame or None
+            DataFrame with sorted values if inplace=False, None otherwise.
+        """
+
+        # Resolve target layer
+        target = []
+        if isinstance(by, str):
+            by = [by]
+        # Check "by" columns for hierarchical references
+        for col in by:
+            if self._is_known_hierarchical_column(col):
+                target.append(col.split(".")[0])
+            else:
+                target.append("base")
+
+        # Ensure one target layer, preventing multi-layer operations
+        target = np.unique(target)
+        if len(target) > 1:
+            raise ValueError("Queries cannot target multiple structs/layers, write a separate query for each")
+        target = str(target[0])
+
+        # Apply pandas sort_values
+        if target == "base":
+            return super().sort_values(
+                by=by,
+                axis=axis,
+                ascending=ascending,
+                inplace=inplace,
+                kind=kind,
+                na_position=na_position,
+                ignore_index=ignore_index,
+                key=key,
+            )
+        elif target in self.nested_columns:
+            target_flat = self[target].nest.to_flat()
+            target_flat = target_flat.set_index(self[target].array.get_list_index())
+
+            if target_flat.index.name is None:  # set name if not present
+                target_flat.index.name = "index"
+            # Index must always be the first sort key for nested columns
+            nested_by = [target_flat.index.name] + [col.split(".")[-1] for col in by]
+
+            # Augment the ascending kwarg to include the index
+            if isinstance(ascending, bool):
+                ascending = [True] + [ascending] * len(by)
+            elif isinstance(ascending, list):
+                ascending = [True] + ascending
+
+            if inplace:
+                target_flat.sort_values(
+                    by=nested_by,
+                    axis=axis,
+                    ascending=ascending,
+                    kind=kind,
+                    na_position=na_position,
+                    ignore_index=False,
+                    key=key,
+                    inplace=True,
+                )
+            else:
+                target_flat = target_flat.sort_values(
+                    by=nested_by,
+                    axis=axis,
+                    ascending=ascending,
+                    kind=kind,
+                    na_position=na_position,
+                    ignore_index=False,
+                    key=key,
+                    inplace=False,
+                )
+            new_df = self._set_filtered_flat_df(nest_name=target, flat_df=target_flat)
```

Comment:
I don't think that we need this filtered helper function here, because we haven't change number of flat rows. I'd do something like following instead:
```python
            new_df = df.copy()
            new_df[target].array[:] = target_flat
```
We probably can optimize it for inplace case.

We do need some `.nest` method for full replacement of the underlying data with a provided flat-frame...

Reply:
oof, sorry. Let's keep your code for now!

Reply:
No worries, will add a comment that we can potentially optimize this